nano/secure/blockstore.cpp

```diff
@@ -87,7 +128,7 @@ bool nano::block_sideband::deserialize (nano::stream & stream_a)
 		boost::endian::big_to_native_inplace (timestamp);
 		if (type == nano::block_type::state)
 		{
-			nano::read (stream_a, epoch);
+			details.deserialize (stream_a);
```

Comment:
You should check the return argument here in case of failure. Or in `block_details::deserialize` don't use a try-catch and let the exception propagate (with an appropriate comment)

Reply:
Oops, good catch. I went with the first approach as we test block_details::deserialize independently.